spec/lib/active_publisher/async/in_memory_adapter_spec.rb

```diff
@@ -0,0 +1,135 @@
+describe ::ActivePublisher::Async::InMemoryAdapter do
+  let(:route) { "test" }
+  let(:payload) { "payload" }
+  let(:exchange_name) { "place" }
+  let(:options) { { :test => :ok } }
+  let(:message) { described_class::Message.new(route, payload, exchange_name, options) }
+  let(:mock_queue) { double(:push => nil, :size => 0) }
+
+  describe "#publish" do
+    before do
+      allow(described_class::Message).to receive(:new).with(route, payload, exchange_name, options).and_return(message)
+      allow(described_class::AsyncQueue).to receive(:new).and_return(mock_queue)
+    end
+
+    it "can publish a message to the queue" do
+      expect(mock_queue).to receive(:push).with(message)
```

Comment:
I think we should move this to be a functional test where we actually get the message back from rabbitmq, but we can do that in a future cut. No reason to block this PR now